Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,224 @@
+        /// </summary>
+        /// <param name="message">The message to print before the prompt.</param>
+        /// <param name="result">The parsing result. Is 0 if the parsing failed.</param>
+        /// <param name="input">The user input.</param>
+        /// <returns>Returns true if the parsing succeeded, false otherwise.</returns>
+        private static bool PromptTryParse(string message, out int result, out string input)
+        {
+            input = PromptInput(message);
+            return int.TryParse(input, out result);
+        }
+
+        /// <summary>
+        /// Prompts for a float value. Converts the string input to a float if possible.
+        /// </summary>
+        /// <param name="message">The message to print before the prompt.</param>
+        /// <param name="result">The parsing result. Is 0 if the parsing failed.</param>
+        /// <param name="input">The user input.</param>
+        /// <returns>Returns true if the parsing succeeded, false otherwise.</returns>
+        private static bool PromptTryParse(string message, out float result, out string input)
+        {
+            input = PromptInput(message);
+            return float.TryParse(input, NumberStyles.Float,
+                CultureInfo.InvariantCulture, out result);
+        }
+
+        /// <summary>
+        /// Prints the message and reads in the user input.
+        /// </summary>
+        /// <param name="message">The message to print.</param>
+        /// <returns>The unparsed and unvalidated input.</returns>
+        private static string PromptInput(string message)
+        {
+            Console.WriteLine(message);
+            return Console.ReadLine();
+        }
+
+        /// <summary>
+        /// Returns a value that indicates the type of input fault./>
+        /// </summary>
+        /// <param name="input">The input the user wrote.</param>
+        /// <returns>An integer indicating the error.</returns>
+        private static int InvalidInput(string input)
+        {
+            return string.IsNullOrEmpty(input)
+                ? EmptyNumberInput
+                : PrintInvalidNumber(input);
+        }
+
+        /// <summary>
+        /// Checks if the given value is in a valid range.
+        /// </summary>
+        /// <param name="value">The value to check.</param>
+        /// <returns>Returns the value if it is in range or a value indicating the opposite.</returns>
+        private static int RangeCheck(int value)
+        {
+            return value > 0
+                ? value
+                : InvalidNumberInput;
+        }
+
+        /// <summary>
+        /// Checks if the given value is in a valid range.
+        /// </summary>
+        /// <param name="value">The value to check.</param>
+        /// <returns>Returns the value if it is in range or a value indicating the opposite.</returns>
+        private static float RangeCheck(float value)
```

Comment:
This doesn't check, this adjusts a number too. Could be a better name.

Reply:
> Sadly that is true. Any suggestions how to name the function?
> Everything I come up with feels bad.

Same issue.